archetypes/servertask/src/main/resources/archetype-resources/src/main/java/Application.java

```diff
@@ -0,0 +1,62 @@
+package ${package};
+
+import java.util.EnumSet;
+import java.util.Set;
+import java.util.concurrent.CompletionStage;
+
+import org.infinispan.commons.configuration.ConfiguredBy;
+import org.infinispan.tasks.ServerTask;
+import org.infinispan.tasks.TaskContext;
+import org.infinispan.tasks.TaskExecutionMode;
+
+import java.nio.charset.StandardCharsets;
+import org.kohsuke.MetaInfServices;
+
+@MetaInfServices
+public class Application<T> implements ServerTask<T> {
```

Comment:
I think something like `CustomServerTask` will be more descriptive as this is not an Application.